statsmodels/stats/control_chart.py

```diff
@@ -0,0 +1,286 @@
+# -*- coding: utf-8 -*-
+"""Control Charts and Statistical Process Control
+
+Created on Sun Dec 31 23:47:52 2017
+
+Author: Josef Perktold
+
+
+Some tools for building control charts and process monitoring.
+The core should be tools for the statistics, estimation and control limits.
+Control charts and monitoring interface are just for a basic example use case.
+
+"""
+
+from __future__ import division
+import numpy as np
+from scipy import stats
+import matplotlib.pyplot as plt
+
+def update_std(std, nobs, y_new):
+    raise NotImplementedError
+
+class Holder(object):
+    def __init__(self, **kwds):
+        self.__dict__.update(kwds)
+
+def control_limits_mean(y0, alpha=0.05, crit=None, distr='t'):
+
+        endog = y0
+        nobs = endog.shape[0]
+        #index = np.arange(nobs)  # not used nor returned
+
+        if crit is not None:
+            crit = crit
+        else:
+            if distr == 't':
+                crit = stats.t.isf(alpha/2, nobs-1)
+            elif distr == 'n':
+                crit = stats.norm.isf(alpha/2)
+            else:
+                raise ValueError('only t and normal for now' )
+
+        center = endog.mean(0)
+        std = endog.std(0, ddof=1)
```

Comment:
center and std computation need to handle weights and use options, e.g. for robust estimators